go/libkb/saltpack_dec.go

```diff
@@ -199,11 +199,25 @@ func (r *TlfKeyResolver) ResolveKeys(identifiers [][]byte) ([]*saltpack.Symmetri
 }
 
 func (r *TlfKeyResolver) getSymmetricKey(info TlfPseudonymInfo) (*saltpack.SymmetricKey, error) {
+	// NOTE: In order to handle finalized TLFs (which is one of the main
+	// benefits of using TLF keys to begin with, for forward readability), we
+	// need the server to tell us what the current, potentially-finalized name
+	// of the TLF is. If that's not the same as what the name was when the
+	// message was sent, we can't necessarily check that the server is being
+	// honest. That's ok insofar as we're not relying on these keys for
+	// authenticity, but it's a drag to not be able to use the pseudonym
+	// machinery.
```

Comment:
Yes, though if there wouldn't be any more verification going on on the client side, is that different from what I'm already doing on `kbweb` to supply the `untrusted_current_name` field?